factory/ethElrondBridgeComponents_test.go

```diff
@@ -344,6 +344,51 @@ func TestEthElrondBridgeComponents_Close(t *testing.T) {
 	})
 }
 
+func TestEthElrondBridgeComponents_startBroadcastJoinRetriesLoop(t *testing.T) {
+	t.Parallel()
+
+	t.Run("close before minTimeBeforeRepeatJoin", func(t *testing.T) {
+		t.Parallel()
+
+		numberOfCalls := 0
+		args := createMockEthElrondBridgeArgs()
+		components, _ := NewEthElrondBridgeComponents(args)
+		components.broadcaster = &testsCommon.BroadcasterStub{
+			BroadcastJoinTopicCalled: func() {
+				numberOfCalls++
```

Comment:
since this will be called asynced because you used a go routine, my suggestion is to use the atomic package when setting/reading this variable. So, L353 will become `numberOfCalls := uint32(0)`, L358 `atomic.AddUint32(&numberOfCalls, 1)`, L368: `assert.Equal(t, uint32(1), atomic.LoadUint32(&numberOfCalls))`
Valid for the next test also.